locality_sensitive_hashing/lsh.py

```diff
@@ -245,17 +326,48 @@ def min_hash(docs_signature_sets: SignatureSet) -> MinHashMatrix:
             Each item (i, j) is the index of of the first shingle, i, that appears in document, j,
             for a given permutation.
     """
+    rng = np.random.default_rng(seed=42)
+
     number_of_permutations = parameters_dictionary["permutations"]
 
     min_hash_signatures: MinHashMatrix = np.empty(
-        shape=(number_of_permutations, docs_signature_sets.shape[1]), dtype=np.int64
+        shape=(number_of_permutations, len(document_list)), dtype=np.int64
     )
 
-    rng = np.random.default_rng(seed=42)
+    # generate unique coefficients for each permutation
+    coefficients = rng.choice(
+        np.arange(total_shingles), size=(number_of_permutations, 2), replace=False
+    )
+
+    # find the next prime number > N
+    if total_shingles % 2 == 0:
+        prime = next_prime(total_shingles + 1)
+    else:
+        prime = next_prime(total_shingles)
+
+    def _hash(x: npt.NDArray[np.intp], p: int, a: int, b: int) -> IntArray:
```

Comment:
Is there a reason for the `_` prefix? Convention when declaring functions inside other functions?

Reply:
Nah, mostly reserved for private functions. Should either move it into the global scope and keep the `_` or just remove  the prefix

Reply:
Renamed in ab17206